tutorials/epochs/plot_40_autogenerate_metadata.py

```diff
@@ -0,0 +1,449 @@
+# Event creation
+# ~~~~~~~~~~~~~~
+#
+# To create epochs with corresponding metadata, we first need to extract all
+# relevant events. In the ERP CORE dataset, experimental events are stored
+# as `~mne.Annotations`. We can convert them to events via
+# `mne.events_from_annotations`:
+
+all_events, all_event_id = mne.events_from_annotations(raw)
+
+###############################################################################
+# Creating metadata from events
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# The basics of ``make_metadata``
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Now it's time to think about the time windows to use for epoching and
+# metadata generation. **It is important to understand that these time windows
+# need not be the same!** That is, the automatically generated metadata might
+# include information about events from only a fraction of the epochs duration;
+# or it might include events that occurred well outside a given epoch.
+#
+# Let us look at a concrete example. In the Flankers task of the ERP CORE
+# dataset, participants were required to respond to visual stimuli by
+# pressing a button. We're interested in looking at the visual evoked responses
+# (ERPs) of trials with correct responses. Based on literature studies, we
+# decide that responses later than 1500 ms after stimulus onset are to be
+# considered invalid, because they don't capture the neuronal processes of
+# interest. We can approach this  in the following way with the help of
+# `mne.epochs.make_metadata`:
+
+# metadata for each epoch shall include events from the range: [0.0, 1.5] s,
+# i.e. starting with stimulus onset and expanding beyond the end of the epoch
+metadata_tmin, metadata_tmax = 0.0, 1.5
+
+# auto-create metadata
+# this also returns a new events array and an event_id dictionary. we'll see
+# later why this is important
+metadata, events, event_id = mne.epochs.make_metadata(
+    events=all_events, event_id=all_event_id,
+    tmin=metadata_tmin, tmax=metadata_tmax, sfreq=raw.info['sfreq'])
+
+# let's look at what we got!
+metadata
+
+###############################################################################
+# Specifying time-locked events
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We can see that the generated table has 802 rows, each one corresponding to
+# an individual event in ``all_events``. The first column, ``event_name``,
+# contains the name of the respective event around which the metadata of that
+# specific column was generated – we'll call that the "time-locked event",
+# because we'll assign it time point zero.
+#
+# Unless you have a very wide screen, you'll most likely have to scroll
+# horizontally to look at the rest of the columns that were automatically
+# created.
+#
+# The names of columns 2 to 7 correspond to the event names specified in the
+# ``all_event_id`` dictionary. These columns contain boolean values – ``True``
+# or ``False`` – indicating whether that specific event occurred during the
```

Comment:
```suggestion
# ``all_event_id`` dictionary. These columns contain boolean values — ``True``
# or ``False`` — indicating whether that specific event occurred during the
```

em-dashes, not en-dashes

Reply:
I thought em-dashes are typically used without spaces around them, though? Which is the main reason for me to use en-dashes when I want to insert (grammatical) parantheses without having to use `(` and `)`

Reply:
em-dash is the correct mark to use for parenthetical comments/asides when you don't want to use `(` and `)`.  Whether to put spaces around the em-dash is, as far as I know, up to the typesetter.  I often see them both with and without surrounding spaces, and I personally think *with* spaces is easier to read.

Reply:
Interesting, I've never seen em-dashes **with** spaces in English books. But yeah, maybe a question of the typesetters' preferences indeed 👌